core/src/main/java/org/infinispan/functional/impl/Params.java

```diff
@@ -108,4 +115,50 @@ public static Params from(Param<?>... ps) {
       }
    }
 
+   public static class Externalizer implements AdvancedExternalizer<Params> {
+      @Override
+      public Set<Class<? extends Params>> getTypeClasses() {
+         return Collections.singleton(Params.class);
+      }
+
+      @Override
+      public Integer getId() {
+         return Ids.PARAMS;
+      }
+
+      @Override
+      public void writeObject(ObjectOutput output, Params object) throws IOException {
+         for (Param param : object.params) {
+            switch (param.serializeMethod()) {
```

Comment:
MetaParams are different thing; I haven't touched those. I understand why these need externalizer per MetaParam.

But as for Params, we control those, so we can handle them in single Params.Externalizer. I'll push update to the other PR, so please let's move the discussion there.